lib/datadog/core/buffer/cruby.rb

```diff
@@ -0,0 +1,59 @@
+# typed: true
+require 'datadog/core/buffer/random'
+
+module Datadog
+  # Buffer that accumulates items for a consumer.
+  # Consumption can happen from a different thread.
+
+  # Buffer that stores objects. The buffer has a maximum size and when
+  # the buffer is full, a random object is discarded.
+  module Core
```

Comment:
This comment should be moved/merged with `class CRuby` below, as right now it's being attached to `Datadog::Core`

(Same note applies elsewhere in the PR)